### ISSUE TITLE
Feature request: allow underscore to mark as unused

### ISSUE THREAD

**Issue by sergeysova** [eslint/eslint#11671]
I propose to add a rule or option, to mark variable or argument as unused and don't treat warning.

Example:

```js
// Here I want to omit body, now I should use eslint-disable-next-line
const { body, ...options } = getOptions()

function someHandler(foo, bar) {
  return bar.run()
}
```

How we can mark variable and argument as unused and disable warning:

```js
const { body: _, ...options } = getOptions()
// or { body: _body,

function someHandler(_foo, bar) {
  return bar.run()
}
```

Inspired by Rust Language. [reference](https://doc.rust-lang.org/book/appendix-02-operators.html#non-operator-symbols)

```rust
fn example(_foo: i32, bar: i32): i32 {
  bar + 1
}

let _parsed = "123".parse();

match foo {
  1 => {},
  2 => {},
  _ => {},
}
```

**Comment by platinumazure** [comment]
Hi @sergeysova, thanks for the issue.

Have you looked at the [`varsIgnorePattern`](https://eslint.org/docs/rules/no-unused-vars#varsignorepattern) and [`argsIgnorePattern`](https://eslint.org/docs/rules/no-unused-vars#argsignorepattern) options in the rule documentation? These should give you the flexibility to configure the rule the way you want.

**Comment by sergeysova** [comment]
@platinumazure wow. Thanks. It absolutely solves my issue
